daml-script/runner/src/main/scala/com/digitalasset/daml/lf/engine/script/Converter.scala

```diff
@@ -376,6 +379,47 @@ object Converter {
     iter(freeAp, List())
   }
 
+  def toDisclosedContract(
+      v: SValue
+  ): Either[String, command.DisclosedContract] =
+    v match {
+      case SRecord(_, _, vals) if vals.size == 3 => {
+        for {
+          cid <- toAnyContractId(vals.get(0))
+          anyKeyM <- toOptional(vals.get(1), toAnyContractKey)
+          anyTemplate <- toAnyTemplate(vals.get(2))
+          keyHashM <- anyKeyM match {
+            case Some(k) =>
+              // TODO (drsk) we probably need to hash the normalized key value here?
+              crypto.Hash.hashContractKey(anyTemplate.ty, k.key.toUnnormalizedValue).map(Some(_))
+            case None => Right(None)
+          }
+        } yield command.DisclosedContract(
+          contractId = cid.coid,
+          templateId = anyTemplate.ty,
+          argument = anyTemplate.arg.toUnnormalizedValue,
+          metadata = command.ContractMetadata(
+            keyHash = keyHashM,
+            createdAt = Time.Timestamp.Epoch, // TODO (drsk) do something better?
```

Comment:
> // TODO (drsk) do something better?

The ledger will reject the transaction if the `createdAt` doesn't match. I'm not familiar with the daml script code, can you somehow use the actual contract metadata returned in the transaction stream?

Reply:
see my comment above on the representation on the Daml side. We really need to use the full metadata here and we need to return it in `query` (and the corresponding variants).